src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -300,22 +300,22 @@ public bool ReadJsonExtensionDataValue(ref ReadStack state, ref Utf8JsonReader r
 
         public Type ParentClassType { get; private set; } = null!;
 
-        public PropertyInfo? PropertyInfo { get; private set; }
+        public MemberInfo? MemberInfo { get; private set; }
 
         public JsonClassInfo RuntimeClassInfo
         {
             get
             {
                 if (_runtimeClassInfo == null)
                 {
-                    _runtimeClassInfo = Options.GetOrAddClass(RuntimePropertyType!);
+                    _runtimeClassInfo = Options.GetOrAddClass(RuntimeMemberType!);
                 }
 
                 return _runtimeClassInfo;
             }
         }
 
-        public Type? RuntimePropertyType { get; private set; } = null;
+        public Type? RuntimeMemberType { get; private set; } = null;
```

Comment:
No need to change this for now. Let's move this up, where "RuntimePropertyType" is.